src/components/Search.js

```diff
@@ -17,33 +12,24 @@ class Search extends Component {
     this.setState({ searchText: e.target.value });
   }
 
-
-
-
   handleSubmit = e => {
     e.preventDefault();
     //console.log("search query value:", this.query.value)
     this.props.onSearch(this.query.value);
     //console.log("search props.onsearch:", this.props.onSearch)
    // console.log('search query value:', this.query.value);
     let query = this.query.value;
-    let path = `/search/${query}`;
-    this.props.history.push(path);
-    e.currentTarget.reset();
-
-  //  console.log('search state:', this.state.searchText);
-   // console.log("search props.onsearch:",this.props.onSearch)
-
-    
-  
+    let path = `/?q=${query}`;
+    this.props.history.push({
+      pathname: '/search',
+      search: path
+    });
+    e.currentTarget.reset();  
```

Comment:
what does this do? We shouldn't be interfacing with the DOM. React does that for us. If you're ever touching HTML in your javascript in react, tahts an anti pattern.